File: Arrays/Algorithms/Sliding_Window/Fixed_Size/average_of_subarrays_of_size_k.py
def average_of_subarrays_of_size_k(arr, k):
	window_sum = sum(arr[:k])
	res = [window_sum / k]
	for i in range(k, len(arr)):
		window_sum += arr[i] - arr[i - k]
		res.append(window_sum / k)
	return res

if __name__ == "__main__":
	n, k = map(int, input().split())
	arr = list(map(int, input().split()))
	print(*[f"{x:.5f}" for x in average_of_subarrays_of_size_k(arr, k)])

File: Arrays/Algorithms/Sliding_Window/Fixed_Size/average_of_subarrays_of_size_k.cpp
#include <bits/stdc++.h>
using namespace std;

vector<double> averageOfSubarraysOfSizeK(const vector<int>& arr, int k) {
	int n = arr.size();
	vector<double> res;
	double window_sum = 0;
	for (int i = 0; i < k; i++) window_sum += arr[i];
	res.push_back(window_sum / k);
	for (int i = k; i < n; i++) {
		window_sum += arr[i] - arr[i - k];
		res.push_back(window_sum / k);
	}
	return res;
}

int main() {
	int n, k;
	cin >> n >> k;
	vector<int> arr(n);
	for (int i = 0; i < n; i++) cin >> arr[i];
	vector<double> res = averageOfSubarraysOfSizeK(arr, k);
	for (double x : res) cout << fixed << setprecision(5) << x << ' ';
	cout << endl;
	return 0;
}
